client/app/components/routes/profile/User.tsx

```diff
@@ -24,7 +24,7 @@ export default async function User() {
   let formData: UserProfilePartialFormData | { error: string } =
     await getUserFormData();
   if ("error" in formData) {
-    if (formData.error.includes("404")) {
+    if (formData.error.includes("Not Found")) {
```

Comment:
@Sepehr-Sobhani , this surprised me a bit--the http error decorator does still return a 404, but when I log `formData` here, I get `{error: Not Found}`. So it seems like the decorator isn't returning exactly the same response as the individual endpoints. This is the first time it's come up--do you think it's a problem?

Reply:
Sorry, I'm not sure if I'm following the issue here. Is there a way I can test this locally?

Reply:
If you pull the branch and leave it as `if (formData.error.includes("404"))`, the e2e profile none test will fail because after the refactor, `formData.error` no longer includes "404". However, as far as I can tell, I am returning a 404 in the decorator. Thanks for your help with this!

Reply:
We previously used `get_object_or_404` to retrieve the user instance, which raised an `Http404` error without any message. Now, we've changed our approach and we still return a `404` error, but it includes a descriptive message. This helps clarify the error's context, making it a more effective solution.